src/Compilers/CSharp/Portable/Parser/LanguageParser_Patterns.cs

```diff
@@ -483,80 +474,19 @@ private bool LooksLikeTupleArrayType()
 
         private PropertyPatternClauseSyntax ParsePropertyPatternClause()
         {
-            ParseSubpatternList(
-                openToken: out SyntaxToken openBraceToken,
-                subPatterns: out SeparatedSyntaxList<SubpatternSyntax> subPatterns,
-                closeToken: out SyntaxToken closeBraceToken,
-                openKind: SyntaxKind.OpenBraceToken,
-                closeKind: SyntaxKind.CloseBraceToken,
-                static p => p.ParseSubpatternElement());
-            return _syntaxFactory.PropertyPatternClause(openBraceToken, subPatterns, closeBraceToken);
-        }
-
-        private void ParseSubpatternList<T>(
-            out SyntaxToken openToken,
-            out SeparatedSyntaxList<T> subPatterns,
-            out SyntaxToken closeToken,
-            SyntaxKind openKind,
-            SyntaxKind closeKind,
-            Func<LanguageParser, T> parseFunc)
-            where T : CSharpSyntaxNode
-        {
-            Debug.Assert((openKind, closeKind) is
-                (SyntaxKind.OpenParenToken, SyntaxKind.CloseParenToken) or
-                (SyntaxKind.OpenBraceToken, SyntaxKind.CloseBraceToken) or
-                (SyntaxKind.OpenBracketToken, SyntaxKind.CloseBracketToken));
-            Debug.Assert(openKind == this.CurrentToken.Kind);
-
-            openToken = this.EatToken(openKind);
-            var list = _pool.AllocateSeparated<T>();
-            try
-            {
-tryAgain:
-                if (this.IsPossibleSubpatternElement() || this.CurrentToken.Kind == SyntaxKind.CommaToken)
-                {
-                    // first pattern
-                    list.Add(parseFunc(this));
-
-                    // additional patterns
-                    int lastTokenPosition = -1;
-                    while (IsMakingProgress(ref lastTokenPosition))
-                    {
-                        if (this.CurrentToken.Kind is SyntaxKind.CloseParenToken or
-                                                      SyntaxKind.CloseBraceToken or
-                                                      SyntaxKind.CloseBracketToken or
-                                                      SyntaxKind.SemicolonToken)
-                        {
-                            break;
-                        }
-                        else if (this.CurrentToken.Kind == SyntaxKind.CommaToken || this.IsPossibleSubpatternElement())
-                        {
-                            list.AddSeparator(this.EatToken(SyntaxKind.CommaToken));
-                            if (this.CurrentToken.Kind is SyntaxKind.CloseBraceToken or SyntaxKind.CloseBracketToken)
```

Comment:
this was particularly heinous in that this was common pattern parsing code that would switch ehre to determine if trailing commas was allowed (for `{` and `[` patterns, but not for `(` patterns).